core/crypto/crypto.c

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <compiler.h>
+#include <kernel/panic.h>
+#include <tee/tee_cryp_provider.h>
+
+#if !defined(_CFG_CRYPTO_WITH_HASH)
```

Comment:
How about making the symbols `__weak` instead? It looks like we could get rid of all the #ifdef-ery in this file.

Reply:
I was thinking the same first, but the problem is that some functions are supposed to be provided in groups and that's not possible with weak functions.